jobsdb/jobsdb.go

```diff
@@ -1525,16 +1504,12 @@ func (jd *HandleT) createDS(appendLast bool, newDSIdx string) dataSetT {
 	_, err = jd.dbHandle.Exec(sqlStatement)
 	jd.assertError(err)
 
-	if appendLast {
-		newDSWithSeqNumber := jd.setSequenceNumber(newDSIdx)
-		jd.JournalMarkDone(opID)
-		return newDSWithSeqNumber
-	}
-
-	//This is the migration case. We don't yet update the in-memory list till
+	//In case of a migration, we don't yet update the in-memory list till
 	//we finish the migration
+	if refreshList {
```

Comment:
I can not easily follow this logic, why `refreshList` leads to `jd.setSequenceNumber(newDS.Index)` being called ?

Reply:
> I can not easily follow this logic, why `refreshList` leads to `jd.setSequenceNumber(newDS.Index)` being called ?

Didn't understand this either. The refactoring's purpose was to leave business logic I couldn't comprehend intact!